app/services/case_extend_deadline_for_sar_service.rb

```diff
@@ -0,0 +1,70 @@
+class CaseExtendDeadlineForSARService
+  attr_reader :result, :error
+
+  def initialize(user, kase, extension_deadline, reason)
+    @user = user
+    @case = kase
+    @extension_deadline = extension_deadline
+    @reason = reason
+    @result = :incomplete
+  end
+
+  def call
+    ActiveRecord::Base.transaction do
+      if validate_params
+        # TODO (Mohammed Seedat): State Machine rules for SAR Extensions
```

Comment:
Think this should be the same as for pit extensions, so managers and approvers can extend in the state drafting for both standard and trigger cases and in the state awaiting dispatch for trigger cases